### DISCUSSION TITLE
Exploring Opportunities and Strategies for CNCF Project Contributions and GSoC 2024 Participation

### DISCUSSION THREAD

**Discussion by SaadBinAbiWaqas** [cncf/mentoring#1093] [Q&A] [answered]
I am a novice enthusiast with a strong desire to participate in CNCF (Cloud Native Computing Foundation) projects, and I aspire to excel in the Google Summer of Code (GSoC) program in 2024. I have some inquiries related to initiating contributions to these projects.
Firstly, what course of action should I take if I decide to contribute to a CNCF project that is not selected for GSoC in 2024?
Secondly, am I at liberty to select any CNCF project at random, and what are the likelihoods of such a project being chosen for participation in the 2024 GSoC program?

**Comment by ghost** [comment]
Hello , can you please help me what to learn so that i can contribute to some project( i've learnt java from kunal )

**Reply by aliok** [reply]
hi @Bunny2104

I would start from here: https://contribute.cncf.io/contributors/getting-started/

**Comment by aliok** [comment, marked as answer]
@SaadBinAbiWaqas 

Thanks for reaching out.

Let me reply inline.

> Firstly, what course of action should I take if I decide to contribute to a CNCF project that is not selected for GSoC in 2024?

We don't know what projects are going to be accepted to GSoC 2024. We don't even know if CNCF will be accepted, or if GSoC 2024 is going to happen. There are no guarantees at anything.

However, I would recommend starting contributing to any project that looks interesting to you and that you feel technically comfortable. Even in the case that the project doesn't participate in GSoC, you will have the knowledge of open source and cloud native tech. This can be very useful when you later apply for GSoC in another project for example.

> Secondly, am I at liberty to select any CNCF project at random, and what are the likelihoods of such a project being chosen for participation in the 2024 GSoC program?

Again, there are no guarantees here. However, you can have a look at previous years' projects to get an idea about which communities are engaging more with GSoC.

**Reply by sadath-12** [reply]
Hi @aliok  how to exactly filter which cncf projects participated in gsoc 2023 ?

**Reply by Nitinshukla88** [reply]
Hey @sadath-12 Visit CNCF ideas list page  in gsoc orgs and there you'll find all the projects which had participated in last year.

**Comment by Nitinshukla88** [comment]
My name's nitin and for this year 2024 I'm also participating in GSOC and interested in engaging with projects of cncf. Hope so I'll be well fitted with all the rules and regulations of community.

**Reply by aliok** [reply]
hi @Nitinshukla88 Thanks for your interest in GSoC at CNCF.

GSoC is announced very recently and we haven't done any work yet. We'll kick start it with the new year.

**Comment by Neon-20** [comment]
Hi Team,
Pranav this Side.

I have built applications for the community previously and shipped them. I am pretty much interested to contribute to cncf this year and also participate in gsoc 2024 as a side product. Looking forward to be at the forefront of grabbing the project-ideas given by cncf or reinvent the wheel in the process. I would love to discuss things in slack as well, but I was unable to find any appropriate channel related to gsoc2024 in cncf. 

Regards
Pranav Rajveer

**Comment by AishwaryJain07** [comment]
Hi @aliok 
This is Aishwary Jain from India.
I am here to get in your Sight regarding GSOC'24
Sir i have good skillset which would be a very good fit for your org. and my experience collaboratively.
my Skillset Includes: Django, C++, Python, C, Javascript, Golang, kubernetes, HTML, CSS, Golang.
kindly assign me a work and on basis of that, give me chance to work with you in GSOC'24.
Thanks.
